File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.js
function solution(myString, pat) {
    var answer = 0;
    
    myString = myString.split("");
    var newString = [];
    
    myString.forEach((e) => {
       if(e === 'A') {
           newString.push('B');
       } else {
           newString.push('A');
       }
    });
    
    newString = newString.join('');
    
    if(newString.indexOf(pat) !== -1) {
        answer = 1;
    }
    
    return answer;
}

File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.java
class Solution {
    public int solution(String myString, String pat) {
        int answer = 0;
        
        String[] strArr = myString.split("");
        StringBuilder newString = new StringBuilder();
        
        for(String s : strArr) {
            if(s.equals("A")) {
                newString.append("B");
            } else {
                newString.append("A");
            }
        }
        
        if(newString.toString().contains(pat)) {
            answer = 1;
        }
        
        return answer;
    }
}